pkg/apis/mxnet/v1alpha2/types.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2018 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha2
+
+import (
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=mxjob
+
+// MXJob represents the configuration of signal MXJob
+type MXJob struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of the MXJob.
+	Spec MXJobSpec `json:"spec,omitempty"`
+
+	// Most recently observed status of the MXJob.
+	// This data may not be up to date.
+	// Populated by the system.
+	// Read-only.
+	Status MXJobStatus `json:"status,omitempty"`
+}
+
+// MXJobSpec is a desired state description of the MXJob.
+type MXJobSpec struct {
+	// CleanPodPolicy defines the policy to kill pods after MXJob is
+	// succeeded.
+	// Default to Running.
+	CleanPodPolicy *CleanPodPolicy `json:"cleanPodPolicy,omitempty"`
```

Comment:
I am not sure if MXNet needs CleanPodRunning, since it is a hack for TensorFlow PS.

Reply:
Actually i met problem when using Running to delete pod , it cannot delete some pods.
I think maybe I should change it to All

Reply:
Running is meant to delete running pods. In TensorFlow, PS are long-running, thus they won't stop. We import CleanPodRunning to allow users to delete these running pods after the job is finished to release the resources.

If MXNet does not have the problem, we could eliminate the field in MXJob.

Reply:
for now , the scheduler\server\worker will be in status "Succeeded" . I don't know for sure if they need the server or scheduler to be long-running so i think we should remain the filed .
I think i can change the default to "All" for now

Reply:
OK, but I think if the servers are succeeded, then MXNet does not have the problem. PS in TensorFlow is designed to be long-running. After the training job is finsihed, PS are still running and consuming resources, this is why we import CleanPodRunning. If servers are succeeded, we can delete it.

But keeping it and changing the default value LGTM.